config/build.js

```diff
@@ -1,29 +1,28 @@
-const alias = require('esbuild-plugin-alias');
+// @ts-check
+
 const path = require('path');
 const pkg = require('../package.json');
 
 const cwd = process.cwd();
 
 const entryPoints = pkg.entries.map((entry) => path.resolve(cwd, entry));
 
-const isProduction = process.env.NODE_ENV === 'production';
-
-module.exports = {
+module.exports = /** @satisfies {import('esbuild').BuildOptions} */ ({
+    alias: Object.fromEntries(
+        Object.entries(pkg.browser).map(([key, value]) => [
+            key,
+            path.resolve(cwd, `${value}.js`),
+        ])
+    ),
     bundle: true,
+    chunkNames: 'chunks/[name]-[hash]',
     entryPoints,
-    format: 'iife',
-    minify: isProduction,
+    entryNames: '[name]',
+    external: ['crypto', 'libxslt'],
+    format: 'esm',
```

Comment:
I meant to call this out in the PR notes, but I can call it out here since it’s also going to cause ESLint failures until I fix the config.

This change will now build for ECMAScript Modules. The reason to make this change now is that it allows the client Transformer to be loaded dynamically only when it’s used. This will have other immediate benefits, by allowing some of the shared code to be loaded concurrently. In the future it’ll allow us to dynamically load quite a bit more, which should significantly reduce load time for many usage scenarios.

This isn’t **technically** a breaking change, as ESM has been supported in our target environments for quite a while, but it’s worth calling out that this is now a hard requirement rather than just a hypothetical one.